src/components/HtmlIdSlugProvider/index.ts

```diff
@@ -0,0 +1,18 @@
+import GithubSlugger from 'github-slugger';
```

Comment:
Placing this script and export the `resetHtmlIdSlugger` in the `astro` file caused the compiler not to be able to find `slugger`. Extracting it to this file solves that issue. But the new problem is that now this script isn't really scoped to the provider anymore.

May need to use a different mechanism, like https://docs.astro.build/en/recipes/sharing-state/ ?

Reply:
Or maybe we could use an actual `context` with this library: https://withastro-utils.github.io/docs/guides/context/